deploy/ansible/roles/iroha-k8s/files/k8s-iroha.yaml

```diff
@@ -29,6 +29,28 @@ spec:
         app: iroha
     spec:
       terminationGracePeriodSeconds: 7
+      initContainers:
+      - name: copy-data
+        image: busybox
+        command: ['sh', '-c', 'cp -v /opt/secrets/$PODNAME/node* /opt/iroha_data && cp -v /opt/iroha_config/* /opt/iroha_data']
+        env:
+          - name: PODNAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+        volumeMounts:
```

Comment:
Ideally that spec to provision additional Iroha peers but right now the `ansible-playbook -e 'replicas=N' playbooks/iroha-k8s/iroha-deploy.yml` can be still be called to provision *N* initial nodes as documented - the k8s-iroha file will be generated with the additional mount points.

In the future I plan to automatically provision new peer secrets for before replicas are created, my current understanding is that I would also need to interact with Iroha and call AddPeer using an authorized account.

Reply:
@bakhtin Yeah the peer volumes are generated in this PR. Automatic peer creation using `addPeer` is not crucial right now. I will look into it when ready :)

Reply:
`AddPeer` command should work, but it requires minimal amount of nodes for completing consensus. So it won't work as a start-up point actually

Reply:
I'm just playing with the idea, nothing decided.

What if we could deploy a Iroha cluster without Ansible. Starting with a minimum set of peers be able to expand the cluster without needing to generate keys explicitly on the remote users machine.

Initially I think this would require a more sophisticated init container with its own docker image.

This process could involve these steps:
0. Generate peer secret (keypair)
1. Spin up a new Iroha pod using this secret (start but wait for peers in genesis.block to be written)
2. > minimal nodes for consensus
  then. Call addPeer
  else. Write to genesis.block using the local config map
3. Repeat until *N* replicas are running

End goal would be to bootstrap a new Iroha cluster with a static k8s spec file?

Reply:
Keep in mind that after peer is added with AddPeer, it cannot be removed from the network. I.e., the cluster can only grow, and is not able to scale down